diff --git a/luci-app-hypermodem/Makefile b/luci-app-hypermodem/Makefile
index ed9530974..38d80b582 100644
--- a/luci-app-hypermodem/Makefile
+++ b/luci-app-hypermodem/Makefile
@@ -17,7 +17,6 @@ LUCI_DEPENDS:=+luci-compat +kmod-usb-net  +kmod-usb-net-cdc-ether +kmod-usb-acm
 		+usbutils \
 		+kmod-qmi_wwan_q \
 		+kmod-pcie_mhi +pciutils \
-		+quectel-CM-5G \
 		+grep \
 
 include $(TOPDIR)/feeds/luci/luci.mk
