diff --git a/luci-app-netdata/Makefile b/luci-app-netdata/Makefile
index e069c4855..4e0a58e1c 100644
--- a/luci-app-netdata/Makefile
+++ b/luci-app-netdata/Makefile
@@ -7,7 +7,7 @@ include $(TOPDIR)/rules.mk
 LUCI_NAME:=luci-app-netdata
 
 LUCI_TITLE:=LuCI Netdata
-LUCI_DEPENDS:=+netdata-ssl +openssl-util
+LUCI_DEPENDS:=+netdata
 
 LUCI_DESCRIPTION:=Real-time performance monitoring
 
diff --git a/netdata/Makefile b/netdata/Makefile
index 3982b3dbb..de798a688 100644
--- a/netdata/Makefile
+++ b/netdata/Makefile
@@ -40,7 +40,7 @@ define Package/netdata
   DEPENDS:=+zlib +libuuid +libuv +libmnl +liblz4 +libjson-c +libyaml +libopenssl +libcurl +protobuf +libzstd +libpcre2
   TITLE:=Real-time performance monitoring tool
   URL:=https://www.netdata.cloud/
-  PROVIDES:=netdata-ssl
+#  PROVIDES:=netdata-ssl
 endef
 
 define Package/netdata/description
