From 1c937095b597cd37cb3865e985fc2725a0376a28 Mon Sep 17 00:00:00 2001
From: chenmozhijin <110812055+chenmozhijin@users.noreply.github.com>
Date: Fri, 23 May 2025 21:20:25 +0800
Subject: [PATCH] fix:no previous prototype for 'fast_classifier_recv'

---
 fast-classifier/src/fast-classifier.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/fast-classifier/src/fast-classifier.c b/fast-classifier/src/fast-classifier.c
index b717017..b804772 100644
--- a/fast-classifier/src/fast-classifier.c
+++ b/fast-classifier/src/fast-classifier.c
@@ -198,6 +198,8 @@ static atomic_t done_fail_msgs = ATOMIC_INIT(0);
  */
 static bool skip_to_bridge_ingress;
 
+int fast_classifier_recv(struct sk_buff *skb);
+
 /*
  * fast_classifier_incr_exceptions()
  *	increase an exception counter.
