diff --git a/openwrt-smartdns/Makefile b/openwrt-smartdns/Makefile
index 4bb350d5..d4be8aaf 100644
--- a/openwrt-smartdns/Makefile
+++ b/openwrt-smartdns/Makefile
@@ -117,7 +117,7 @@ define Download/smartdns-webui
 	FILE:=$(SMARTDNS_WEBUI_FILE)
 	PROTO:=$(SMARTDNS_WEBUI_SOURCE_PROTO)
 	URL:=$(SMARTDNS_WEBUI_SOURCE_URL)
-	MIRROR_HASH:=42e280cbfc168f88690652b85ce919c2aeedc9ef7c4b55122742e484b23492cd
+	MIRROR_HASH:=skip
 	VERSION:=$(SMARTDNS_WEBUI_SOURCE_VERSION)
 	HASH:=$(SMARTDNS_WEBUI_HASH)
 	SUBDIR:=smartdns-webui
