diff --git a/openwrt-smartdns/Makefile b/openwrt-smartdns/Makefile
index 1f2d02ef..997d67c7 100644
--- a/openwrt-smartdns/Makefile
+++ b/openwrt-smartdns/Makefile
@@ -46,7 +46,7 @@ endef
 define Package/smartdns
   $(Package/smartdns/default)
   TITLE:=smartdns server
-  DEPENDS:=+libpthread +libopenssl +libatomic
+  DEPENDS:=+libpthread +libopenssl +libatomic +zlib
 endef
 
 define Package/smartdns/description
